riscv/csrs.cc

```diff
@@ -979,6 +980,17 @@ reg_t time_counter_csr_t::read() const noexcept {
 
 void time_counter_csr_t::sync(const reg_t val) noexcept {
   shadow_val = val;
+  if (proc->extension_enabled(EXT_SSTC)) {
+    auto timer_val = state->stimecmp->read();
+    // do nothing if timer is 0
+    if (timer_val && shadow_val >= timer_val) {
```

Comment:
Is there a case where `timer_val` has wrapped around to 0, so we should turn off STIP?